### PR TITLE
Fix imitation

### PR DESCRIPTION
**Pull request by takerfume**

### CODE REVIEW

**Comment by rarilurelo** [comment]
What did you change?

File: example/run_airl.py [hunk 1]
```diff
@@ -205,9 +203,10 @@
 
 if args.pretrain:
     with measure('bc pretrain'):
-        _ = behavior_clone.train(
-            expert_traj, pol, optim_pol, args.bc_batch_size, args.bc_epoch
-        )
```
Comment (takerfume) [inline example/run_airl.py]: This was bug, `behavior_clone.train` doesn't have the argument of `epoch`.
Therefore, it is a bug that `args.bc_epoch` is passed to `behavior_clone.train`.

File: machina/loss_functional.py [hunk 2]
```diff
@@ -656,30 +656,18 @@ def cross_ent_diayn(discrim, batch, num_skill):
...
-    discrim_loss = F.binary_cross_entropy_with_logits(
-        logits, torch.ones(len, device=get_device())*expert_or_agent)
-    return discrim_loss
-
-
-def density_ratio_adv_cross_ent(advf, pol, batch, expert_or_agent):
-    obs = batch['obs']
-    acs = batch['acs']
-    advs, _ = advf(obs, acs)
-    energies = advs
```
Comment (takerfume) [inline machina/loss_functional.py]: Previously, I defined two function `density_ratio_rew_cross_ent` and `density_ratio_adv_cross_ent`.
But the difference between these is how to calculate `energies`, so I have combined this.

File: example/run_airl.py [hunk 3]
```diff
@@ -236,29 +235,29 @@
...
                                      optim_pol=optim_pol,
                                      clip_param=args.clip_param, max_grad_norm=args.max_grad_norm, gamma=args.gamma)
 
         else:
             result_dict = airl.train(agent_traj, expert_traj, pol, vf, optim_vf, optim_discrim,
-                                     rewf=rewf, shaping_vf=shaping_vf, advf=advf, rew_type=args.rew_type,
+                                     rewf=rewf, shaping_vf=shaping_vf, advf=advf,
                                      rl_type=args.rl_type,
-                                     pol_ent_beta=args.pol_ent_beta, discrim_ent_beta=args.discrim_ent_beta,
+                                     pol_ent_beta=args.pol_ent_beta,
```
Comment (takerfume) [inline example/run_airl.py]: I have removed unnecessary argument `rew_type` and `discrim_ent_beta`

File: example/run_airl.py [hunk 4]
```diff
@@ -205,9 +203,10 @@
 
 if args.pretrain:
     with measure('bc pretrain'):
-        _ = behavior_clone.train(
-            expert_traj, pol, optim_pol, args.bc_batch_size, args.bc_epoch
-        )
+        for _ in range(args.bc_epoch):
+            _ = behavior_clone.train(
+                expert_traj, pol, optim_pol, args.bc_batch_size
+            )
```
Comment (takerfume) [inline example/run_airl.py]: This was bug, behavior_clone.train doesn't have the argument of epoch.
Therefore, it is a bug that args.bc_epoch is passed to behavior_clone.train.
Then, I have fixed.

**Comment by takerfume** [comment]
@rarilurelo I have written the comments about what I changed. Could you check them?

**Pull request merged**